src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/run.py

```diff
@@ -67,25 +75,48 @@ def acr_run(cmd,  # pylint: disable=too-many-locals
         source_location = check_remote_source_code(source_location)
         logger.warning("Sending context to registry: %s...", registry_name)
 
-    platform_os, platform_arch, platform_variant = get_validate_platform(cmd, os_type, platform)
-
-    FileTaskRunRequest, PlatformProperties = cmd.get_models('FileTaskRunRequest', 'PlatformProperties')
-    request = FileTaskRunRequest(
-        task_file_path=file,
-        values_file_path=values,
-        values=(set_value if set_value else []) + (set_secret if set_secret else []),
-        source_location=source_location,
-        timeout=timeout,
-        platform=PlatformProperties(
-            os=platform_os,
-            architecture=platform_arch,
-            variant=platform_variant
-        ),
-        credentials=get_custom_registry_credentials(
-            cmd=cmd,
-            auth_mode=auth_mode
+    if file == "-":
+        import sys
+        YAML_TEMPLATE = ""
+        for s in sys.stdin.readlines():
+            YAML_TEMPLATE += s
+        values_content = ""
+
+    if cmd_value:
+        YAML_TEMPLATE = "steps: \n  - cmd: {{ .Values.run_image }}\n"
+        values_content = "run_image: {0}\n".format(cmd_value)
+
+    if values_content is not None:
```

Comment:
Should you check YAML_TEMPLATE as well?

Reply:
Was thinking the existence of values_content infers the existence of yaml_template. Not sure if it is the other around. To sum up the discussions on Teams channel, for now RP supports either task_content + values_content or task_content + values. Might make more sense to check yaml_template instead of values_content.